sequence_task_1/student/include/impl.h

```diff
@@ -9,32 +9,65 @@ class StaticPlaylist
 public:
     /** @todo Member traits */
 
+	using value_type = typename Container::value_type;
+	using reference = typename Container::reference;
+	using const_reference = typename Container::const_reference;
+	using iterator = typename Container::iterator;
+	using const_iterator = typename Container::const_iterator;
+	using difference_type = typename Container::difference_type;
+	using size_type = typename Container::size_type;
+
     /** @todo Iterators */
 
+	const_iterator begin() const {return m_tracklist.cbegin();}
+	const_iterator end() const {return m_tracklist.cend();}
+
+
     StaticPlaylist() = default;
 
     /** @todo Constructor from any reversible sequence container */
 
+	template <typename reversable_sequence_container>
+	StaticPlaylist(reversable_sequence_container init) {
```

Comment:
reversable_sequence_container will be the container itself, so here you're doing the copy - please don't, add the const reference!